src/components/graphs/ScatterGraph.vue

```diff
@@ -0,0 +1,499 @@
+<template>
+  <div :id="containerId" ref="chartContainer" class="scatterplot-container">
+    <div v-if="loading" class="loading-overlay">
+      <div class="loading-spinner"></div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { onMounted, onBeforeUnmount, ref, watch, computed } from 'vue';
+import * as d3 from 'd3';
+import { DataPoint, RegressionLine } from '../citywide-stats/types';
+
+interface ChartElements {
+  container: d3.Selection<HTMLElement, unknown, null, undefined>;
+  chartGroup: d3.Selection<SVGGElement, unknown, null, undefined>;
+  tooltip: d3.Selection<HTMLDivElement, unknown, null, undefined>;
+  circles: d3.Selection<SVGCircleElement, DataPoint, SVGGElement, unknown>;
+  path: d3.Selection<SVGPathElement, DataPoint[], null, undefined>;
+  trendLine: d3.Selection<SVGLineElement, unknown, null, undefined> | null;
+}
+
+const props = withDefaults(
+  defineProps<{
+    data: DataPoint[];
+    yAxisLabel: string;
+    color?: string;
+    containerId: string;
+    title: string;
+    showGrid: boolean;
+    showTrendLine?: boolean;
+    animationDuration?: number;
+  }>(),
+  {
+    color: '#3B82F6',
+    showGrid: true,
+    showTrendLine: true,
+    animationDuration: 800,
+  },
+);
+
+const tooltipAnimationDuration = 300;
+const chartContainer = ref<HTMLElement | null>(null);
+const loading = ref(true);
+const hasAnimated = ref(false);
+const chartRendered = ref(false);
+let xScale: d3.ScaleLinear<number, number> = d3.scaleLinear().range([0, 0]);
+let yScale: d3.ScaleLinear<number, number> = d3.scaleLinear().range([0, 0]);
+
+// Store references to chart elements for later animation
+let chartElements: ChartElements | null = null;
+
+const sortedData = computed(() => {
+  return [...props.data].sort((a, b) => a.year - b.year);
+});
+
+const renderChartStructure = (): void => {
```

Comment:
I'm a little confused by the structure of this file - why is this not a `class` like the other component files? It seems like this is basically a file just grouping together some functions, which doesn't seem ideal.